### ISSUE TITLE
Missing annotations

### ISSUE THREAD

**Issue by cyberalien** [milesj/emojibase#42]
Some skins of emojis are missing annotations in `/en/data.json` (from NPM package) and also seem to be missing in `packages/data/en/raw.json` in this repository:
```
1F9D1-1F3FB-200D-1F91D-200D-1F9D1-1F3FC
1F9D1-1F3FB-200D-1F91D-200D-1F9D1-1F3FD
1F9D1-1F3FB-200D-1F91D-200D-1F9D1-1F3FE
1F9D1-1F3FB-200D-1F91D-200D-1F9D1-1F3FF
1F9D1-1F3FC-200D-1F91D-200D-1F9D1-1F3FD
1F9D1-1F3FC-200D-1F91D-200D-1F9D1-1F3FE
1F9D1-1F3FC-200D-1F91D-200D-1F9D1-1F3FF
1F9D1-1F3FD-200D-1F91D-200D-1F9D1-1F3FE
1F9D1-1F3FD-200D-1F91D-200D-1F9D1-1F3FF
1F9D1-1F3FE-200D-1F91D-200D-1F9D1-1F3FF
1F469-1F3FB-200D-1F91D-200D-1F469-1F3FC
1F469-1F3FB-200D-1F91D-200D-1F469-1F3FD
1F469-1F3FB-200D-1F91D-200D-1F469-1F3FE
1F469-1F3FB-200D-1F91D-200D-1F469-1F3FF
1F469-1F3FC-200D-1F91D-200D-1F469-1F3FD
1F469-1F3FC-200D-1F91D-200D-1F469-1F3FE
1F469-1F3FC-200D-1F91D-200D-1F469-1F3FF
1F469-1F3FD-200D-1F91D-200D-1F469-1F3FE
1F469-1F3FD-200D-1F91D-200D-1F469-1F3FF
1F469-1F3FE-200D-1F91D-200D-1F469-1F3FF
1F468-1F3FB-200D-1F91D-200D-1F468-1F3FC
1F468-1F3FB-200D-1F91D-200D-1F468-1F3FD
1F468-1F3FB-200D-1F91D-200D-1F468-1F3FE
1F468-1F3FB-200D-1F91D-200D-1F468-1F3FF
1F468-1F3FC-200D-1F91D-200D-1F468-1F3FD
1F468-1F3FC-200D-1F91D-200D-1F468-1F3FE
1F468-1F3FC-200D-1F91D-200D-1F468-1F3FF
1F468-1F3FD-200D-1F91D-200D-1F468-1F3FE
1F468-1F3FD-200D-1F91D-200D-1F468-1F3FF
1F468-1F3FE-200D-1F91D-200D-1F468-1F3FF
```

**Comment by milesj** [comment]
@cyberalien Not all emojis have annotations, and it looks like the skin tone variations do not: https://github.com/unicode-org/cldr/blob/release-36/common/annotations/en.xml#L889

That being said, it should inherit from the parent.

**Comment by cyberalien** [comment]
I'm a bit confused. In XML file you've linked to, none of emojis have skin tones. It only lists default emojis without skin tones.

Emojis with mixed skin do have annotations in .json files, for example `1F9D1-1F3FC-200D-1F91D-200D-1F9D1-1F3FB`, yet just 2 entries below item `1F9D1-1F3FC-200D-1F91D-200D-1F9D1-1F3FD` has no annotation. The only difference between those items is order of skin modifiers. So if one entry has annotation, shouldn't other entry have annotation as well?

**Comment by milesj** [comment]
@cyberalien I have to piece all this together manually, so I probably just missed this case.

**Comment by cyberalien** [comment]
Oh, I see. I assumed it was automated, which would mean there is a small bug in your script.

But since its manual, I've added those entries and made a pull request: https://github.com/milesj/emojibase/pull/43

Looks like pull request fails because it doesn't follow some format, which I don't see documentation for, so you can just pick changes from it and re-add with your own commit. Saves you a bit of work 😀

**Comment by milesj** [comment]
It is automated, but I have to manually piece it all together in the generator. The unicode/emoji spec is very complex and they don't really provide the data in a nice format.

The parent inheritance happens here: https://github.com/milesj/emojibase/blob/master/packages/generator/src/generators/generateData.ts#L98 But it looks like the formats are different for these.

**Comment by milesj** [comment]
So the data does exist in the XML, but somehow it's being lost. https://github.com/unicode-org/cldr/blob/release-36/common/annotationsDerived/en.xml#L2830

**Comment by milesj** [comment]
Ok, I think I know the problem, but I'm not sure how to easily fix it. 

Let's use "Men Holding Hands: Medium Skin Tone, Medium-Dark Skin Tone" (`1F468-1F3FD-200D-1F91D-200D-1F468-1F3FE`) as an example, which has a missing annotation in raw.  https://emojipedia.org/emoji/%F0%9F%91%A8%F0%9F%8F%BD%E2%80%8D%F0%9F%A4%9D%E2%80%8D%F0%9F%91%A8%F0%9F%8F%BE/

This unicode character DOES NOT exist in the annotations XML file: https://github.com/unicode-org/cldr/blob/release-36/common/annotationsDerived/en.xml#L2830 But if you reverse the sequence, it does exist under "men holding hands: medium-dark skin tone, medium skin tone" instead of "men holding hands: medium skin tone, medium-dark skin tone".

So to summarize, the hexcode up top does not exist, but its reversed variant does `1F468-1F3FE-200D-1F91D-200D-1F468-1F3FD`. Hence why a handful of them are missing annotations.

I'm not sure if this is by design and they expect consumers to handle the permutations ourselves... which is annoying. But at least I know what the problem is now.

**Comment by milesj** [comment]
Added a work around for now. This will be fixed in the next patch version.
